acs-integration-tests/src/test/java/com/ge/predix/integration/test/ExternalResourceAttributeReaderIT.java

```diff
@@ -0,0 +1,137 @@
+            + "\"value\":\"13a71359-db68-4602-aac5-a8fa401c3194\"}]";
+
+    @Autowired
+    private RedisTemplate<String, String> redisTemplate;
+
+    @Autowired
+    private AbstractAttributeCache attributeCache;
+
+    @Autowired
+    @Spy
+    private ExternalResourceAttributeReader externalResourceAttributeReader;
+
+    @BeforeClass
+    public void beforeClass() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testGetAttributes() throws Exception {
+        Answer<Set<Attribute>> answer = invocation -> {
+            String identifier = (String) invocation.getArguments()[1];
+
+            // TODO: Switch to using AdapterResponse once the US77483 branch is merged into develop
+            String response = new RestTemplate().getForObject(ASSET_ADAPTER_URI + identifier, String.class);
+            Map<String, Object> responsePojo = OBJECT_MAPPER
+                    .readValue(response, new TypeReference<Map<String, Object>>() {
+                    });
+            List<Map<String, String>> attributeList = (List<Map<String, String>>) responsePojo.get(ATTRIBUTES);
+
+            Set<Attribute> attributes = new HashSet<>();
+            for (Map<String, String> attributeMap : attributeList) {
+                attributes
+                        .add(new Attribute(attributeMap.get(ISSUER), attributeMap.get(NAME), attributeMap.get(VALUE)));
+            }
+
+            return attributes;
+        };
+
+        Mockito.doAnswer(answer).when(this.externalResourceAttributeReader).getAttributesFromAdapter(ZONE, RESOURCE_ID);
+        Mockito.doReturn(ZONE).when(this.externalResourceAttributeReader).getZoneId();
+
+        Set<Attribute> attributes = this.externalResourceAttributeReader.getAttributes(RESOURCE_ID);
+
+        List<Map<String, String>> aircraftPart = OBJECT_MAPPER
+                .readValue(ASSET_ATTRIBUTES_JSON, new TypeReference<List<Map<String, String>>>() {
+                });
+        Map<String, String> singleAircraftPart = aircraftPart.get(0);
+
+        for (Attribute attribute : attributes) {
+            Assert.assertTrue(singleAircraftPart.get(attribute.getName()).equals(attribute.getValue()));
+        }
+
+        String resourceKey = AbstractPolicyEvaluationCache.resourceKey(ZONE, RESOURCE_ID);
+
+        Assert.assertTrue(StringUtils.isNotEmpty(this.redisTemplate.opsForValue().get(resourceKey)));
+
+        String resourceAttributesKey = this.attributeCache.resourceAttributesKey(ZONE, RESOURCE_ID);
+
+        Assert.assertEquals(resourceAttributesKey, resourceKey + ':' + ATTRIBUTES);
+        Assert.assertEquals(this.redisTemplate.opsForValue().get(resourceAttributesKey), ADAPTER_ATTRIBUTES_JSON);
+
+        Long ttl = this.redisTemplate.getExpire(resourceAttributesKey, TimeUnit.SECONDS);
+        Assert.assertTrue(ttl > 0L && ttl <= this.attributeCache.getConfiguredTtl());
+    }
+}
```

Comment:
We need to add another IT: PolicyEvalCachingWithExternalResourceAttributeReaderIT to assert that policy decision is reevaluated if attributes cache is expired and attributes are refreshed.